sklearn/impute/tests/test_impute.py

```diff
@@ -1710,3 +1710,59 @@ def test_simple_imputer_keep_empty_features(strategy, array_type, keep_empty_fea
             assert_array_equal(constant_feature, 0)
         else:
             assert X_imputed.shape == (X.shape[0], X.shape[1] - 1)
+
+
+@pytest.mark.parametrize("csc_container", CSC_CONTAINERS)
+def test_imputation_custom(csc_container):
+    X = np.array(
+        [
+            [1.1, 1.1, 1.1],
+            [3.9, 1.2, np.nan],
+            [np.nan, 1.3, np.nan],
+            [0.1, 1.4, 1.4],
+            [4.9, 1.5, 1.5],
+            [np.nan, 1.6, 1.6],
+        ]
+    )
+
+    X_true = np.array(
+        [
+            [1.1, 1.1, 1.1],
+            [3.9, 1.2, 1.1],
+            [0.1, 1.3, 1.1],
+            [0.1, 1.4, 1.4],
+            [4.9, 1.5, 1.5],
+            [0.1, 1.6, 1.6],
+        ]
+    )
+
+    imputer = SimpleImputer(
+        missing_values=np.nan,
+        strategy="custom",
+        sparse_fn=np.min,
+        dense_fn=np.ma.min,
```

Comment:
Just calling out that I think there's a foot-gun here for most users in knowing that it's necessary to use a dense_fn of `np.ma.min` instead of `np.min`, and that that exposes the internals of SimpleImputer. For my use-case it doesn't matter much (I already know that I need to use the np.ma variant for dense_fn), but I wager there'll be some support load from this.